lib/xnvme_be_linux_async_liburing.c

```diff
@@ -80,6 +84,24 @@ xnvme_be_linux_liburing_init(struct xnvme_queue *q, int opts)
 	// Ring-initialization
 	//
 	if (queue->poll_sq) {
+		if (!g_sqpoll_ring_initialized) {
```

Comment:
I was wondering if we need a lock for this?

Reply:
Good point. I am wavering a bit between adding a lock or just adding a comment on  ``xnvme_queue_init()`` saying that it is not thread-safe. Since, currently none of the backends guard the ring/qpair/control-block/etc. initializations.
However, a quick skim indicates that they do not need to... so I guess the best thing would be to add a lock here, so be on the safe side, and then go over the implementations ensuring that they are thread-safe, fix them, and then note the state.

Reply:
Locking added and improved error-handling.